.github/workflows/pages.yml

```diff
@@ -0,0 +1,35 @@
+name: Push to gh-pages
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Get dependencies
+      run: |
+        cd ./emulator
+        curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+        wasm-pack build -t web
+        cd ../emulator-webapp
+        npm install --legacy-peer-deps
+    - name: Build docs
+      run: |
+        cd ./emulator-webapp
+        npm run build-docs
+    - name: Commit files
```

Comment:
Would it theoretically be possible to create this commit on a seperate branch (not main) so the main history doesn't get cluttered?

Reply:
I believe the new version does exactly that: It configures git, creates the docs and pushes them to the branch `gh-pages`. Is that how you've imagined it?

Right now the commit is created on `main` and pushed to `gh-pages`. I'm not sure if I've overseen anything by doing so, that might cause us trouble regarding `main`s history.

Reply:
Sounds good